frontend/checks-runner.ts

```diff
@@ -0,0 +1,204 @@
+import checkbox from '@inquirer/checkbox';
+import chalk from 'chalk';
+import { spawn } from 'child_process';
+import fs from 'fs/promises';
+import ora from 'ora';
+import type { Ora } from 'ora';
+import path from 'path';
+
+// Types
+interface PackageJson {
+  scripts?: Record<string, string>;
+}
+
+interface Script {
+  value: string;
+  name: string;
+  description: string;
+  checked?: boolean;
+  args?: string;
+  emoji?: string;
+  useRawOutput?: boolean; // New flag for scripts that need raw output
+}
+
+type AllowedScripts = readonly Script[];
+
+// Configuration with emojis
+const allowedScripts: AllowedScripts = [
+  {
+    value: 'format:check',
+    name: 'Prettier',
+    description: 'Opinionated code formatter',
+    emoji: '✨',
+  },
+  {
+    value: 'lint:check',
+    name: 'ESLint',
+    description: 'Statically analyzes your code to find problems',
+    emoji: '🔍',
+  },
+  {
+    value: 'typecheck',
+    name: 'Typescript',
+    description: 'Run compiler by checking your code',
+    emoji: '📝',
+  },
+  {
+    value: 'build',
+    name: 'Build',
+    description: 'Build application and server',
+    emoji: '📦',
+  },
+  {
+    value: 'test',
+    name: 'Vitest',
+    description: 'Run next generation unit testing framework',
+    args: 'run',
+    emoji: '🧪',
+  },
+  {
+    value: 'test:e2e',
+    name: 'Playwright',
+    description: 'Run reliable end-to-end testing for modern web apps',
+    args: '--quiet',
+    emoji: '🎭',
+  },
+];
+
+class ScriptRunner {
+  private currentProcess: ReturnType<typeof spawn> | null = null;
+  private spinner: Ora;
+
+  constructor() {
+    this.spinner = ora();
+    this.setupProcessHandlers();
+  }
+
+  private setupProcessHandlers(): void {
+    const shutdown = () => {
+      console.log(chalk.yellow('\n👋 Gracefully shutting down...'));
+      if (this.currentProcess) {
+        console.log(chalk.yellow('🛑 Terminating running scripts...'));
+        this.currentProcess.kill('SIGTERM');
+        this.currentProcess = null;
```

Comment:
This line is unnecessary since you're immediately calling `process.exit()`

Reply:
Which line?

Reply:
Line 83.. setting `currentProcess` to `null`.

Reply:
Line has been removed. 🫡